code/modules/asset_cache/assets/rtd.dm

```diff
@@ -0,0 +1,117 @@
+		/obj/item/stack/tile/iron/white/herringbone,
+		/obj/item/stack/tile/iron/white_side,
+		/obj/item/stack/tile/iron/white_corner,
+		/obj/item/stack/tile/iron/cafeteria,
+		/obj/item/stack/tile/iron/white/textured,
+		/obj/item/stack/tile/iron/white/textured_edge,
+		/obj/item/stack/tile/iron/white/textured_half,
+		/obj/item/stack/tile/iron/white/textured_corner,
+		/obj/item/stack/tile/iron/white/textured_large,
+		/obj/item/stack/tile/iron/recharge_floor,
+		/obj/item/stack/tile/iron/smooth,
+		/obj/item/stack/tile/iron/smooth_edge,
+		/obj/item/stack/tile/iron/smooth_half,
+		/obj/item/stack/tile/iron/smooth_corner,
+		/obj/item/stack/tile/iron/smooth_large,
+		/obj/item/stack/tile/iron/terracotta,
+		/obj/item/stack/tile/iron/terracotta/small,
+		/obj/item/stack/tile/iron/terracotta/diagonal,
+		/obj/item/stack/tile/iron/terracotta/herringbone,
+		/obj/item/stack/tile/iron/kitchen,
+		/obj/item/stack/tile/iron/kitchen/small,
+		/obj/item/stack/tile/iron/kitchen/diagonal,
+		/obj/item/stack/tile/iron/kitchen/herringbone,
+		/obj/item/stack/tile/iron/chapel,
+		/obj/item/stack/tile/iron/showroomfloor,
+		/obj/item/stack/tile/iron/solarpanel,
+		/obj/item/stack/tile/iron/freezer,
+		/obj/item/stack/tile/iron/grimy,
+		/obj/item/stack/tile/iron/sepia,
+		/obj/item/stack/tile/glass,
+		/obj/item/stack/tile/rglass,
+		/obj/item/stack/tile/circuit,
+		/obj/item/stack/tile/circuit/green,
+		/obj/item/stack/tile/circuit/red,
+	)
+
+	for(var/obj/item/stack/tile/tile as anything in tiles)
+		var/icon_state = initial(tile.icon_state)
+		Insert(sprite_name = icon_state , I = 'icons/obj/tiles.dmi', icon_state = icon_state)
+
+		var/list/tile_directions = GLOB.tile_rotations[initial(tile.singular_name)]
+		if(tile_directions == null)
+			continue
+
+		for(var/direction as anything in tile_directions)
+			var/icon/img = icon(icon = 'icons/obj/tiles.dmi', icon_state = icon_state)
+			switch(direction)
+				if(NORTH)
+					img.Turn(-180)
+				if(WEST)
+					img.Turn(90)
+				if(EAST)
+					img.Turn(-90)
+				if(NORTHEAST)
+					img.Turn(-180)
+					var/icon/east_rotated = icon(icon = 'icons/obj/tiles.dmi', icon_state = icon_state)
+					east_rotated.Turn(-90)
+					img.Blend(east_rotated,ICON_MULTIPLY)
+					img.SetIntensity(2,2,2)
+				if(NORTHWEST)
+					img.Turn(-180)
+					var/icon/west_rotated = icon(icon = 'icons/obj/tiles.dmi', icon_state = icon_state)
+					west_rotated.Turn(90)
+					img.Blend(west_rotated,ICON_MULTIPLY)
+					img.SetIntensity(2,2,2)
+				if(SOUTHEAST)
+					var/icon/east_rotated = icon(icon = 'icons/obj/tiles.dmi', icon_state = icon_state)
+					east_rotated.Turn(-90)
+					img.Blend(east_rotated,ICON_MULTIPLY)
+					img.SetIntensity(2,2,2)
+				if(SOUTHWEST)
+					var/icon/west_rotated = icon(icon = 'icons/obj/tiles.dmi', icon_state = icon_state)
+					west_rotated.Turn(90)
+					img.Blend(west_rotated,ICON_MULTIPLY)
+					img.SetIntensity(2,2,2)
+			Insert(sprite_name = "[icon_state]-[dir2text(direction)]", I = img)
```

Comment:
Agh, this is a lot of directions, and icon creation is not at all cheap. Ideally we would be doing this styling in the CSS instead, but I don't know if IE supports that well.

Reply:
Actually since we already can use transform I'm fairly sure we can. Please try and see if you can get that to work.

Reply:
Actually I need ur advice on this.
I can create an image which will automatically orient the icon in any specified direction using image() function but I can't convert it into an icon for the sprite sheet.

How do I convert an image into an icon?

Reply:
I'm a bit confused.

The way I would expect this to work is that this sends the tile image for every single tile, and then the client uses `rotate` in `transform` to perform the rotation based on direction.

You shouldn't need `image` at all and you can just use `icon` for sending the initial icon.

Reply:
Rotate accepts angle in degrees as the parameter not directions likes NORTH, SOUTH etc.

Reply:
plz stop marking these as resolved just because you think they can't be helped, this is a seriously pressing issue that needs my actual say on